### ISSUE TITLE
Accessibility: aria support for paginator

### ISSUE THREAD

**Issue by danielcollettepb** [primefaces/primeng#6812]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
When tabbing through the paging controls with a screen reader, the first/previous/next/last buttons do not read text. Tabbing to the items per page dropdown does not show focus state.

**Expected behavior**
The screen reader should speak (vi aria-label) when tabbed to each control "First", "Previous", "Next", "Last". When tabbing to the items per page dropdown a focus state should be shown to indicate the current control.

**Minimal reproduction of the problem with instructions**
![image](https://user-images.githubusercontent.com/6116867/47897650-0143ce80-de49-11e8-8694-994d16b96c92.png)

**What is the motivation / use case for changing the behavior?**
Support for screen reader accessibility

* **Angular version:** 6

* **PrimeNG version:** 6.1.5

* **Browser:** MacOS: Chrome, Firefox, Safari

**Comment by tharahan** [comment]
We are facing the same problem in PrimeNG 11.x. Any ETA on when will this be fixed?

**Comment by rueeggl** [comment]
same here with PrimeNG 13.0.4. The fix would be very much appreciated.

**Comment by komby** [comment]
We are seeing this error in our Accessibility testing, also using 13.0.4.  We also verified that the error occurs on https://www.primefaces.org/primeng/showcase/#/table/filter  when using the Axe tool it reports  
`{ 
    "ruleId": "button-name",
    "description": "Ensures buttons have discernible text",
    "help": "Buttons must have discernible text",
    "helpUrl": "https://dequeuniversity.com/rules/axe/4.3/button-name?application=AxeChrome",
    "impact": "critical",
    "isManual": false,
    "selector": ["#pr_id_3 > p-paginator > .p-paginator-bottom.p-paginator.p-component > .p-paginator-first.p-disabled.p-paginator-element"],
    "summary": "Fix any of the following:\n  Element does not have inner text that is visible to screen readers\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Element has no title attribute\n  Element's default semantics were not overridden with role=\"none\" or role=\"presentation\"",
    "source": "<button type=\"button\" pripple=\"\" class=\"p-ripple p-element p-paginator-first p-paginator-element p-link p-disabled ng-star-inserted\" disabled=\"\"><span class=\"p-paginator-icon pi pi-angle-double-left\"></span><span class=\"p-ink\"></span></button>",
    "tags": ["cat.name-role-value", "wcag2a", "wcag412", "section508", "section508.22.a", "ACT"],
    "igt": "",
    "shareURL": "",
    "createdAt": "2022-02-03T19:03:39.748Z",
    "testUrl": "https://www.primefaces.org/primeng/showcase/",
    "foundBy": ""
}`

Is there any known workarounds that can be used for this?

**Comment by S4You** [comment]
Hi, we need this fix too. What is the timeline for this fix? Any known temporary solution?

**Comment by darKaz3** [comment]
Hi everyone.

I also needed this functionallity for my work, especially because everything I do has to be accessible, but since I am not one to wait I implemented it myself. I will share with those of you who have a need to also have this ASAP in your applications and I am available to further clarifications. Keep in mind that I did this for Angular.

So my first step was to identify how to tackle the issue. For that I went the good old way of querying the DOM. Not the prettiest thing to do, but in this case, I was out of options. Once I had the elements, I iterated through them and added the aria-label attribute as required.

I packed my class in a neat Utils file:
`@Injectable()
export class AriaLabeler{
    //Method to add aria-label attribute and text to all paginator 
    //elements of the primeNg table.
    public addsAriaLabel2Paginators(): void {
        //list of common classes in the paginator elements
        const tempList = ["p-paginator-last", "p-paginator-first", "p-paginator-next", "p-paginator-prev"]
        //find all paginator link elements
        const linkElList = Array.from(document.querySelectorAll("button.p-paginator-element"));
        linkElList.forEach(x => {
          //clears classList of extra classes
          x.classList.remove("p-ripple", "p-element", "p-paginator-element", "p-link");
                  
          //adds aria label if it is a page
          if (x.textContent) {
              x.classList.contains("p-highlight") ?
                  x.setAttribute("aria-label", "Page" + x.textContent + ", current page") :
                  x.setAttribute("aria-label", "Page " + x.textContent)
            
                    
            //adds aria-label for remaining pagination elements
          } else if (tempList.includes(x.classList[0])) {
            x.setAttribute("aria-label", "Navigate to page " + this.getClassTranslation(x.classList[0]))
          }
          //repopulates classList with deleted classes
          x.classList.add("p-ripple", "p-element", "p-paginator-element", "p-link");
        });
      }
    
    //Method that returns the correct DE name of the navigation elements in pagination.
    private getClassTranslation(className: string) {
        if (className == "p-paginator-last") return "last";
        else if (className == "p-paginator-first") return "first";
        else if (className == "p-paginator-next") return "next";
        else if (className == "p-paginator-prev") return "previous";
        else return "";
    }
}`

Which I then added to the providers in my app.module:
`...
providers: [
...
AriaLabeler
]
...`

And finally I added it to my app.component, since I have several tables spread across several components and I abhorr duplicate code. If you only needed it for one component, you can add it just there. Also, I added to ngAfterViewChecked. This does make more calls, but it is the safest way to ensure that everytime your table changes the aria-labels get updated. The alternative is to call the method on your subscribe, once you get the data. Here is how my version to use in the whole application in app.component looks like:
`  //Called after every check of the component's view. Applies to components only.
  public ngAfterViewChecked(): void {
    //Adds aria-label to the pagination elements in all tables.
    this.ariaUtil.addsAriaLabel2Paginators();
  }`

For other elements in the paginator, you just need to extrapolate from what I've done, I only do need it for the basic ones, but the guidelines are there. This is my fix until the accessibility changes come live.
Cheers!

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you reopen the issue so we can include it in our roadmap? 
Please don't forget to add your feedback as a comment after reopening the issue. These will be taken into account by us and will contribute to the development of this feature. Thanks a lot for your understanding!

Best Regards,
